### DISCUSSION TITLE
Maxing CPU when running health checks

### DISCUSSION THREAD

**Discussion by Jamesking56** [spatie/laravel-health#167] [Q&A]
Hi,

I'm using an AWS server with my Laravel app inside of Docker. I've scheduled the health checks every minute according to the documentation and its causing max CPU spikes every minute on the server:

<img width="1440" alt="Screenshot 2023-03-09 at 3 30 05 pm" src="https://user-images.githubusercontent.com/253237/224073111-4e3f6a31-6c04-4251-89c7-06e1312cae36.png">

Why does this package use so much CPU power and so many processes?

**Comment by freekmurze** [comment]
I'm not seeing this problem in any projects where I use the package. I'm assuming it is caused by something specific in your projects / servers.

Should you find that the package is in error, feel free to submit a PR.

**Reply by Jamesking56** [reply]
I registered these checks in my `AppServiceProvider` `boot` method, is this correct?

```php
public function boot(): void
    {
        Health::checks([
            UsedDiskSpaceCheck::new(),
            DatabaseCheck::new(),
            CacheCheck::new(),
            OptimizedAppCheck::new(),
            DebugModeCheck::new(),
            QueueCheck::new(),
            ScheduleCheck::new(),
        ]);

...
```

**Reply by freekmurze** [reply]
Seems correct
